src/components/CurrentUser/CurrentUser.jsx

```diff
@@ -1,12 +1,74 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
 import './CurrentUser.scss';
 
-export const CurrentUser = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+import { getUser } from '../../api/api';
+import { currentUserType } from '../../types';
 
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+export class CurrentUser extends PureComponent {
+  state = {
+    user: {},
+  }
+
+  componentDidMount() {
+    this.updateUser(this.props.userId);
+  }
+
+  componentDidUpdate() {
+    const { userId } = this.props;
+
+    this.updateUser(userId);
+  }
```

Comment:
compare userId in prevProps and this.props and call state by condition

https://reactjs.org/docs/react-component.html#componentdidupdate

>If I select another user the details should be updated. (use componentDidUpdate).
>If I select the same user there should not be a request to the server.